test/check-machines-create

```diff
@@ -1788,6 +1817,41 @@ vnc_password= "{vnc_passwd}"
         self.assertIn("<os>", domainXML)
         self.assertNotIn("efi", domainXML)
 
+    def testCreateDownloadRhel(self):
+        runner = TestMachinesCreate.CreateVmRunner(self, rhel_download=True)
+        config = TestMachinesCreate.TestCreateConfig
+
+        b = self.browser
+        m = self.machine
+
+        self.login_and_go("/machines")
+        b.wait_in_text("body", "Virtual machines")
+
+        m.execute(f"qemu-img create {TestMachinesCreate.TestCreateConfig.VALID_DISK_IMAGE_PATH} 500M")
+
+        runner.createTest(TestMachinesCreate.VmDialog(self, sourceType='os',
+                                                      os_name=config.RHEL_5_0,
+                                                      os_short_id=config.RHEL_5_0_SHORTID,
+                                                      offline_token="my_offline_token",
+                                                      create_and_run=False))
+
+        m.execute("rm /var/lib/libvirt/images/rhel-5-0-boot.iso")
+
+        runner.createTest(TestMachinesCreate.VmDialog(self, sourceType='os',
+                                                      os_name=config.RHEL_5_0,
+                                                      os_short_id=config.RHEL_5_0_SHORTID,
+                                                      offline_token="my_offline_token",
+                                                      create_and_run=True))
+
+        # If run on different connection, the downloaded iso should be stored in ~/.local/share/libvirt/images/
+        runner.createTest(TestMachinesCreate.VmDialog(self, sourceType='os',
+                                                      os_name=config.RHEL_5_0,
+                                                      os_short_id=config.RHEL_5_0_SHORTID,
+                                                      offline_token="my_offline_token",
+                                                      create_and_run=True,
+                                                      connection="session"),
+                          check_env_empty=False)
```

Comment:
Can you please add a test what happens if you specify a wrong token, or the download fails? There should be an error message in the dialog, and the possibility to fix/retry.